### DISCUSSION TITLE
RSpec Controller Tests and Rodauth

### DISCUSSION THREAD

**Discussion by martinstreicher** [jeremyevans/rodauth#240] [General]
What is the proper technique to login a user to run an RSpec controller spec?

**Comment by martinstreicher** [comment]
```
      session[:account_id] = account.id
      session[:authenticated_by] = 'password'
      get :new
      expect(response).to be_ok
```

Using the Googler, I found mention of adding session values, but this does not seem to work.

**Comment by jeremyevans** [comment]
Not sure what an controller test would represent in this case. Rodauth is build on Roda, which doesn't have the equivalent of a "controller".  In general, it's best to test applications using Rodauth with a tool such as capybara.  See Rodauth's test suite for examples.

**Comment by martinstreicher** [comment]
The answer above does indeed work. I will post an example in a moment. An account must also be verified.

**Comment by martinstreicher** [comment]
I created an `Account` in Rodauth with this: 

```
FactoryBot.define do
  factory :account do
    email    { Faker::Internet.email }
    password { 'password' }
    status   { 'verified' }
  end
end
```

and then in the RSpec controller spec, wrote this: 

```
it '...' do
  session[:account_id] = account.id
  session[:authenticated_by] = 'password'
  get :new
  expect(response).to redirect_to(activity_question_path(Activity.last, Activity.last.current_question))
end
```
